src/Codec/Xlsx/Templater.hs

```diff
@@ -105,30 +68,35 @@ fixRowHeights rh r n = insertCopies $ shift removeOriginal
       Just h -> foldr (\x m' -> M.insert x h m') m [r..(r + n -1)]
       Nothing -> m
 
-
-runSheet :: Xlsx -> Int -> (TemplateDataRow, TemplateSettings, [TemplateDataRow]) -> IO Worksheet
-runSheet x n (cdr, ts, d) = do
-  ws <- sheet x n
-  let
-    templateRows = if tsOrientation ts == Columns then transpose $ toList ws else toList ws
-    repeatRow = tsRepeated ts
-    (prolog, templateRow : epilog) = splitAt repeatRow templateRows
-    tpl = buildTemplate repeatRow templateRow
-    prolog' = replacePlaceholders prolog cdr
+applyTemplateOnSheet :: (TemplateDataRow, TemplateSettings, [TemplateDataRow]) -> Worksheet -> Worksheet
+applyTemplateOnSheet (cdr, ts, d) ws =
+    ws & wsColumns           .~ cw
+       & wsRowPropertiesMap  .~ rh
+       & wsCells             %~ applyTemplateOnCellMap
+  where
+    mayTranspose =
```

Comment:
what do you think about `fixOrientation`?
Other idea is to move `Columns` into this function parameter, so pattern matching will be a part of its declaration and not a nested `case` which makes code somewhat harder to read